core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -229,9 +259,12 @@ private[spark] class Executor(
       // ClosedByInterruptException during execBackend.statusUpdate which causes
       // Executor to crash
       Thread.interrupted()
+      notifyAll()
```

Comment:
Basically, my concern here was that if I did optimize to just use `notify()` then we'd risk bugs if someone else came along later and added more `wait()`ing threads.

And actually I think we might be able to wind up with two waiting threads in the case where you first cancel with interrupt=False and then cancel again with interrupt=True... an example of why it's hard to reason about whether only `notify()` is correct.